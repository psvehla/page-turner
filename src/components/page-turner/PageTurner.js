import React from 'react';

import {
  PAGE_ONE,
  PAGE_TWO,
  PAGE_THREE,
  PAGE_FOUR,
  PAGE_FIVE,
  PAGE_SIX,
  PAGE_SEVEN,
} from '../../pages.js';

import PageOne from '../page-one/PageOne';
import PageTwo from '../page-two/PageTwo';
import PageThree from '../page-three/PageThree';
import PageFour from '../page-four/PageFour';
import PageFive from '../page-five/PageFive';
import PageSix from '../page-six/PageSix';
import PageSeven from '../page-seven/PageSeven';

import './PageTurner.css';

function PageTurner(props) {
  switch (props.currentPage) {
    case PAGE_ONE:
      return (
        <PageOne onClick={props.onClick} />
      );
    case PAGE_TWO:
      return (
        <PageTwo onClick={props.onClick} />
      );
    case PAGE_THREE:
      return (
        <PageThree onClick={props.onClick} />
      );
    case PAGE_FOUR:
      return (
        <PageFour onClick={props.onClick} />
      );
    case PAGE_FIVE:
      return (
        <PageFive onClick={props.onClick} />
      );
    case PAGE_SIX:
      return (
        <PageSix onClick={props.onClick} />
      );
    case PAGE_SEVEN:
      return (
        <PageSeven onClick={props.onClick} />
      );
    default:
      return (
        <div>
          <p>Current page not recognised!</p>
          <PageOne onClick={props.onClick} />
        </div>
      );
  }
}

export default PageTurner;
