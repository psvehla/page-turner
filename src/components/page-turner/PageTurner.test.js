import React from 'react';
import renderer from 'react-test-renderer';
import PageTurner from './PageTurner';

it('renders correctly', () => {

  const tree = renderer
    .create(<PageTurner />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
