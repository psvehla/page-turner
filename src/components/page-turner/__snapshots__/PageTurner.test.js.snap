// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Current page not recognised!
  </p>
  <div>
    <img
      alt="bike"
      onClick={[Function]}
      src="1.jpg"
    />
  </div>
</div>
`;
